packages/mattermost-redux/src/types/client4.ts

```diff
@@ -2,3 +2,21 @@
 // See LICENSE.txt for license information.
 
 export * from '@mattermost/types/client4';
+
+type ErrorOffline = {
+    message: string;
+    url: string;
+};
+type ErrorInvalidResponse = {
+    intl: {
+        id: string;
+        defaultMessage: string;
+    };
+};
+export type ErrorApi = {
+    message: string;
+    server_error_id: string;
+    status_code: number;
+    url: string;
+};
+export type Client4Error = ErrorOffline | ErrorInvalidResponse | ErrorApi;
```

Comment:
Something similar to this might belong in `@mattermost/client`, but there's some weird conflicts between this type, the `ClientError` type that's in `@mattermost/client`, and the `ServerError` type that, as the name says, comes from the server.